addons/website/models/website_menu.py

```diff
@@ -94,7 +94,7 @@ def create(self, vals):
 
     def write(self, values):
         res = super().write(values)
-        if 'website_id' in values or 'group_ids' in values:
+        if 'website_id' in values or 'group_ids' in values or 'sequence' in values:
```

Comment:
not sure to understand :/
Only the ids is cached, so the sequence should not changes the id :/

Don't you think that the right fix will be to re-order the menu instead ?

Reply:
we could, we would remove this clear_caches and the clear_caches in `unlink` since `sorted` remove the elements that have been deleted

since the original commit seemed to be about decreasing query count ( https://github.com/odoo/odoo/pull/47257 ) I think clear_caches should be alright.